src/services/auth/plugins/roles/preHandlers.ts

```diff
@@ -0,0 +1,25 @@
+import { FastifyReply, FastifyRequest, RouteHandlerMethod } from 'fastify';
+
+import { InsufficientPermission } from '../../error';
+import { RoleStrategy } from './types';
+
+/**
+ * preHandler to check if user validate one of the preconditions to access a route.
+ * @param strategies One or more role strategies to check if user validate the preconditions.
+ * @throws {InsufficientPermission} If user does not validate any of the preconditions.
+ * @throws {GraaspAuthError} If only one role strategy is provided and it failed with a provided error.
+ */
```

Comment:
It's really a suggestion, you can skip it if you want ;)

```suggestion
/**
 * Pre-handler function that checks if the user meets at least one of the specified access preconditions.
 * @param strategies The array of role strategies to check for access.
 * @throws {InsufficientPermission} If user does not satisfy any of the preconditions.
 * @throws {GraaspAuthError} If only one role strategy is provided and it failed with a provided error.
 */
```